qldbdriver/qldbdriver.go

```diff
@@ -171,8 +176,9 @@ func (driver *QLDBDriver) GetTableNames(ctx context.Context) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		tableNames := make([]string, 0)
-		for result.Next(txn) {
```

Comment:
Why is this change here? Removing txn from Next breaks a very important concept.

Reply:
Because the implementation of `result.Next(txn)` doesn't actually do anything with `txn` beyond pass it into the recursive call of `result.Next(txn)`.
And also because BufferedResult has a `Next() bool` method defined allowing that method to be part of the Result interface.

Reply:
Added txn back to Next() and updated Result interface accordingly.